packages/cli/src/util/logger.ts

```diff
@@ -1,47 +1,88 @@
+  createWinstonLogger,
   TimestampFormat,
   TimestampFormatCode,
+  LogFormat,
+  logFormats,
 } from "@lodestar/utils";
+import {ConsoleDynamicLevel} from "./loggerConsoleTransport.js";
 
 export const defaultLogMaxFiles = 5;
 
 export interface ILogArgs {
   logLevel?: LogLevel;
   logFileLevel?: LogLevel;
-  logFormatGenesisTime?: number;
-  logFormatId?: string;
   logFileDailyRotate?: number;
-}
-
-export function errorLogger(): ILogger {
-  return new WinstonLogger({level: LogLevel.error});
+  logFormatGenesisTime?: number;
+  logPrefix?: string;
+  logFormat?: string;
+  logLevelModule?: string[];
 }
 
 /**
  * Setup a CLI logger, common for beacon, validator and dev commands
  */
-export function getCliLogger(args: ILogArgs, paths: {logFile?: string}, config: IChainForkConfig): ILogger {
-  const transports: TransportOpts[] = [{type: TransportType.console}];
+export function getCliLogger(
+  args: ILogArgs,
+  paths: {logFile?: string},
+  config: IChainForkConfig,
+  opts?: {hideTimestamp?: boolean}
+): ILogger {
+  const consoleTransport = new ConsoleDynamicLevel({
+    // Set defaultLevel, not level for dynamic level setting of ConsoleDynamicLevel
+    defaultLevel: args.logLevel ?? LogLevel.info,
+    debugStdout: true,
+    handleExceptions: true,
+  });
+
+  if (args.logLevelModule) {
+    for (const logLevelModule of args.logLevelModule ?? []) {
+      const [module, levelStr] = logLevelModule.split("=");
+      const level = levelStr as LogLevel;
+      if (LogLevel[level as LogLevel] === undefined) {
+        throw Error(`Unknown level in logLevelModule '${logLevelModule}'`);
+      }
+
+      consoleTransport.setModuleLevel(module, level);
+    }
+  }
+
+  const transports: TransportStream[] = [consoleTransport];
+
   if (paths.logFile) {
-    transports.push({
-      type: TransportType.file,
-      filename: paths.logFile,
-      level: args.logFileLevel,
-      // yargs populates with undefined if just set but with no arg
-      // $ ./bin/lodestar.js beacon --logFileDailyRotate
-      // args = {
-      //   logFileDailyRotate: undefined,
-      // }
-      rotate: "logFileDailyRotate" in args,
-      maxfiles: args.logFileDailyRotate ?? defaultLogMaxFiles,
-    });
+    // yargs populates with undefined if just set but with no arg
+    // $ ./bin/lodestar.js beacon --logFileDailyRotate
+    // args = {
+    //   logFileDailyRotate: undefined,
+    // }
+    const rotate = "logFileDailyRotate" in args;
+    const filename = paths.logFile;
+
+    transports.push(
+      rotate
+        ? new DailyRotateFile({
+            level: args.logFileLevel,
+            //insert the date pattern in filename before the file extension.
+            filename: filename.replace(/\.(?=[^.]*$)|$/, "-%DATE%$&"),
+            datePattern: "YYYY-MM-DD",
+            handleExceptions: true,
+            maxFiles: args.logFileDailyRotate ?? defaultLogMaxFiles,
+            auditFile: path.join(path.dirname(filename), ".log_rotate_audit.json"),
```

Comment:
For dockerized setups the audit file was placed in the process root and lost on restart. So DailyRotateFile could not clean the log files.